sklearn/decomposition/_sparse_pca.py

```diff
@@ -361,6 +380,29 @@ class MiniBatchSparsePCA(SparsePCA):
     0.94
     """
 
+    _parameter_constraints = {
+        k: v
+        for k, v in SparsePCA._parameter_constraints.items()
+        if k
+        in [
+            "n_components",
+            "alpha",
+            "ridge_alpha",
+            "verbose",
+            "n_jobs",
+            "method",
+            "random_state",
+        ]
+    }
+    _parameter_constraints.update(
```

Comment:
This thing indicates us that both `SparsePCA` and `MiniBatchSparsePCA` should have a common base class.
We can improve this in another PR.